### ISSUE TITLE
web 用 socket io 1.7 连接不上问题.WebSocketTransport

### ISSUE THREAD

**Issue by hymmyh** [mrniko/netty-socketio#557]
因为调用了下面 close的代码,导致连接不上. 
       if (client != null && client.isTransportChannel(ctx.channel(), Transport.WEBSOCKET)) {
            log.debug("channel inactive {}", client.getSessionId());
            client.onChannelDisconnect();
        }
        super.channelInactive(ctx);
        if (client != null) {
            client.send(packet);
        }
        channel.close();
        ctx.close();

把代码改成

      if (client != null && client.isTransportChannel(ctx.channel(), Transport.WEBSOCKET)) {
            log.debug("channel inactive {}", client.getSessionId());
            client.onChannelDisconnect();
            client.send(packet);
            channel.close();
            ctx.close();
        }
        super.channelInactive(ctx);

**Comment by jkohjk** [comment]
this is correct. the channel should not be closed if it is not supposed to be disconnected. but i am wondering what is the point of sending the packet? can it / should it send a packet after it has already called onChannelDisconnect?
